wp/sections/feature.tex

```diff
@@ -0,0 +1,75 @@
+\par As described above, diKTat is very configurable and user-friendly. But these are not all of its advantages and features. Below will be presented and described unusual and important killer-feature diKTat.
+
+\subsection{Configuration file}
+\par
+It's worth starting with the configuration files. This is a file in which the user can manually turn rules on and off or configure the rules settings. Below is one of the rules in the configuration file.
+\begin{center}
+- name: FILE\underline{ }IS\underline{ }TOO\underline{ }LONG\\
+	enabled: true\\
+	configuration:\\
+	maxSize: '2000'\\
+	ignoreFolders: ' '\\
+\end{center}
+Each rule in this file consists of 3 parts: name - the name of the rule, enabled - the ability to enable or disable the rule, configuration - parameters for the rules. With the first two, everything is obvious. The third parameter is less obvious. The configuration is passed "properties" to check in this rule. For example, if you have a rule "FILE\underline{ }IS\underline{ }TOO\underline{ }LONG" - checking the number of lines in a Kotlin file, then the user can configure the maximum number of lines in the file - by changing the "maxSize" in the configuration,or the user can specify paths to folders that do not need to be checked - by writing the path before these daddy in "ignoreFolders". \\
+
+\subsection{Create ASTNode}
+\par
+Another fuature is a method that allows you to construct a syntax tree from text. Here's an example of how it works:
+
+\begin{figure}[H]
+  \centering
+  \includegraphics[scale=0.5]{wp/pictures/code.png}
+  \caption{Example how to call this method}   
+\end{figure}
+
+\begin{figure}[H]
+  \centering
+  \includegraphics[scale=0.5]{wp/pictures/tree.png}
+  \caption{The tree we get}   
+\end{figure}
+
+As you can see in the examples, we pass to the method the text of the nodes that we want to receive and the flag, which is set to false by default. The flag should be set to true in order to immediately build a tree with a root node of the FILE type. What's going on inside this method? First of all, the system properties are set (for example: set "idea.io.use.nio2" to true). Further in the method, the text is checked. If the text of the code contains such keywords as import or package, then the method builds a tree with a root node of the FILE type, otherwise it tries with a different root type. In both cases, at the end, if the tree contains an ERROR\underline{ }ELEMENT type of node, it means that an error was made in the code and the method was unable to build the tree and, therefore, throws an exception.\\
```

Comment:
This algorithm can parse the code even partially, when you do not need to save the hierarchy of the file (with imports/packages/classes).

For example it can parse and provide you a sub-tree for these lines of code:
```
if(true) {
println("Hello world")
}
```

This helps us to implement such complex inspections like the detection of commented code, helps easily fix the code without manually building sub-trees in visitors.